Source/Icebreaker/Services/MatchingService.cs

```diff
@@ -204,9 +255,40 @@ private string GetChannelUserObjectId(ChannelAccount account)
             this.Randomize(users);
 
             var pairs = new List<Tuple<ChannelAccount, ChannelAccount>>();
-            for (int i = 0; i < users.Count - 1; i += 2)
+
+            var currentPairs = new HashSet<string>();
+
+            for (int i = 0; i < users.Count - 1; i++)
             {
-                pairs.Add(new Tuple<ChannelAccount, ChannelAccount>(users[i], users[i + 1]));
+                for (int j = i + 1; j < users.Count; j++)
+                {
+                    if (currentPairs.Contains(users[i].Id) || currentPairs.Contains(users[j].Id))
```

Comment:
NIT: you use these value `users[i]` and `users[j]` multiple times. It would aid in clarity if you assigned them to a variable at the top of this inner loop such as 

```csharp
var user1 = users[i];
var user2 = users[j];
```
or even just having `user1Id` and `user2Id` since the rest of the user object is omitted